kratos/controllers/controller.h

```diff
@@ -0,0 +1,175 @@
+  @author Daniel Diez
+*/
+class Controller
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of Controller
+    KRATOS_CLASS_POINTER_DEFINITION(Controller);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    explicit Controller() = default;
+
+    /// Destructor.
+    virtual ~Controller() {}
+
+    /// Copy constructor.
+    //TODO: Check. It is required by the registry
+    Controller(Controller const& rOther) {}
+
+    /// Move constructor
+    Controller(Controller&& rOther) = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /// Assignment operator.
+    Controller& operator=(Controller const& rOther) = delete;
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+     * @brief This method creates an pointer of the controlle
+     * @details We consider as input a Model and a set of Parameters for the sake of generality
+     * @warning Must be overrided in controller implementation
+     * @param rModel The model to be consider
+     * @param ThisParameters The configuration parameters
+     */
+    virtual Controller::Pointer Create(
+        Model& rModel,
+        Parameters ThisParameters) const
+    {
+        KRATOS_ERROR << "Calling base class Create. Please override this method in the corresonding Controller" << std::endl;
+        return nullptr;
+    }
+
+    /**
+     * @brief Checks that input conditions are correct. 
+     */
+    virtual int Check()
+    {
+        return 0;
+    }
+
+    /**
+     * @brief Using input data, returns bool.
+     */
+    virtual bool Evaluate()
+    {
+        KRATOS_ERROR << "Calling base class Evaluate. Please override this method in the corresonding Controller" << std::endl;
+    }
+
+    /**
+     * @brief This method provides the defaults parameters to avoid conflicts between the different constructors
+     */
+    virtual const Parameters GetDefaultParameters() const
+    {
+        KRATOS_ERROR << "Calling the base Controller class GetDefaultParameters. Please implement the GetDefaultParameters in your derived process class." << std::endl;
+        const Parameters default_parameters = Parameters(R"({})" );
+
+        return default_parameters;
+    }
+
+    ///@}
+    ///@name Access
+    ///@{
+
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const
+    {
+        return "Controller";
+    }
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const
+    {
+        rOStream << "Controller";
+    }
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const
+    {
+    }
+
+    ///@}
+}; // Class Controller
+
+///@name Type Definitions
+///@{
+
+
+///@}
+///@name Input and output
+///@{
+
+/// input stream function
+inline std::istream& operator >> (std::istream& rIStream, Controller& rThis);
```

Comment:
```suggestion
std::istream& operator >> (std::istream& rIStream, Controller& rThis);
```

Reply:
Its also like this in process and operation, but I think its better to remove it as you suggest. There is clearly an overuse of "inline" in kratos code.